syntax = "proto3";

service Manager {
  rpc Evaluation (EvaluationRequest) returns (EvaluationReply) {}
}

message EvaluationRequest {
  string challenge_name = 1;
  string code = 2;
}

message EvaluationReply {
  string result = 1;
}

service SandboxExec {
  rpc Exec (ExecRequest) returns (ExecReply) {}
  rpc WriteStdin (WriteStdinRequest) returns (WriteStdinReply) {}
  rpc Terminate (TerminateRequest) returns (TerminateReply) {}
}

message ExecRequest {
  string code = 1;
  string options = 2;
}

message ExecReply {
  string result = 1;
}

message WriteStdinRequest {
  string id = 1;
  string content = 2;
}

message WriteStdinReply {
  string result = 1;
}

message TerminateRequest {
  string id = 1;
}

message TerminateReply {
  string output = 1;
}